c++/src/kj/string.h

```diff
@@ -504,7 +648,44 @@ String strArray(T&& arr, const char* delim) {
   return result;
 }
 
-template <typename... Params>
+enum class StringifyContext {
```

Comment:
I kind of feel like having two separate functions `strPreallocated()` and `strSignalSafe()` would be cleaner and easier to understand, rather than have a template parameter specify the behavior.